src/CabalHelper/Compiletime/Compat/Environment.hs

```diff
@@ -1,6 +1,19 @@
+{-# LANGUAGE CPP #-}
 module CabalHelper.Compiletime.Compat.Environment where
 
-import System.Environment
+import qualified System.Environment (getEnvironment)
+#ifndef mingw32_HOST_OS
+import qualified System.Posix.Env (setEnv)
+#endif
 
 lookupEnv :: String -> IO (Maybe String)
-lookupEnv var = do env <- getEnvironment; return (lookup var env)
+lookupEnv var =
+  do env <- System.Environment.getEnvironment
+     return (lookup var env)
+
+setEnv :: String -> String -> IO ()
+#ifdef mingw32_HOST_OS
+setEnv = System.Environment.setEnv
```

Comment:
Again this can't (and probably doesn't) compile since you don't import setEnv in the import for `System.Environment` above.